PathPlanning/TimeBasedPathPlanning/SafeInterval.py

```diff
@@ -0,0 +1,303 @@
+"""
+Safe interval path planner
+    This script implements a safe-interval path planner for a 2d grid with dynamic obstacles. It is faster than
+    SpaceTime A* because it reduces the number of redundant node expansions by pre-computing regions of adjacent
+    time steps that are safe ("safe intervals") at each position. This allows the algorithm to skip expanding nodes
+    that are in intervals that have already been visited earlier.
+
+    Reference: https://www.cs.cmu.edu/~maxim/files/sipp_icra11.pdf
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+from PathPlanning.TimeBasedPathPlanning.GridWithDynamicObstacles import (
+    Grid,
+    Interval,
+    ObstacleArrangement,
+    Position,
+    empty_2d_array_of_lists,
+)
+import heapq
+import random
+from dataclasses import dataclass
+from functools import total_ordering
+import time
+
+# Seed randomness for reproducibility
+RANDOM_SEED = 50
+random.seed(RANDOM_SEED)
+np.random.seed(RANDOM_SEED)
```

Comment:
I think the random seed should be set either in the main function that runs the simulation or within the __init__ function of PathPlanner. What do you think?

Reply:
Moved into the init function here:
https://github.com/AtsushiSakai/PythonRobotics/pull/1184/commits/21b363d840ff63a4cb367d268248fe8a2a1ef550#diff-bf9f23426e59ae4a03e19fbac049e7ba85684f6108698d6b27ff5b84894359cfR107